crates/iota-sdk-graphql-client/src/query_types/transaction.rs

```diff
@@ -126,12 +147,24 @@ pub struct TxBlockEffects {
     pub effects: Option<TransactionBlockEffects>,
 }
 
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema = "rpc", graphql_type = "TransactionBlock")]
+pub struct TxBlockCheckpoint {
```

Comment:
The naming is really annoying here, `TxBlockCheckpoint` vs `TransactionBlockCheckpoint` vs `TransactionBlockCheckpointQuery`... Let's try to explicit what they actually do and how they differ

Reply:
I'm just mimicking the other query, and tbh I'm not sure there's much we can do to make this better. These queries are bound to have weird names because of the nature of `cynic` requiring so many structs

Reply:
I disagree, we can probably derive a sensible name from the `graphql_type` + fields we are trying to fetch. Having `TxBlockCheckpoint` and `TransactionBlockCheckpoint` is not acceptable. But this probably means changing the whole file to harmonise so I'll open an issue for later.

Reply:
https://github.com/iotaledger/iota-rust-sdk/issues/360